### PR TITLE
[fix] Change foreign key attributes to LazyReferenceField

### PR DESCRIPTION
**Pull request by SammyBaek**
## Changes made
- This PR changes the attribute types of foreign keys in MongoDB from `ObjectId` to `LazyReferenceField`.
  - This lets MongoEngine auto-fetch the reference objects of the foreign keys without querying ourselves (which will get bloated real fast with our nested hierarchy).
  - ref: https://docs.mongoengine.org/apireference.html#mongoengine.fields.LazyReferenceField
- NB: Lazy Reference only has access to `_id`. Need to call `fetch()` in order to retrieve other fields of the object.

## Testing
- [x] End-to-End
Ensure that current data in DB returns `LazyReference` with no migration.
Screen Shot 2021-11-09 at 10 37 27 PM

### CODE REVIEW

File: bin/dev [hunk 1]
```diff
@@ -68,15 +68,15 @@ runWebpack(){
 # Run pytest
 runPytest(){
     findBackendContainerId container_id
-    cmd="docker exec -t "$container_id" pytest "$*
+    cmd="docker exec -it "$container_id" pytest "$*
```
Comment (SammyBaek) [inline bin/dev:71]: allows interaction with shell i.e. debuggers

File: server/tests/model/test_user.py [hunk 2]
```diff
@@ -119,12 +115,11 @@ def test_starter_family_ids(mocker: MockerFixture) -> None:
     assert len(family_ids) == len(STARTER_FAMILIES) + 1  # account for unassigned family
 
     found_unassigned_family = False
-    for family_id in family_ids:
-        family = Family.lookup_family(family_id)
-        assert family
-        assert family.get_id() == str(family_id)
+    for family_ref in family_ids:
+        family = family_ref.fetch()
+        assert family.get_id() == str(family.id)
```
Comment (EpicPi) [inline server/tests/model/test_user.py:120]: don't think this check is necessary anymore

File: server/tests/model/test_user.py [hunk 3]
```diff
@@ -100,12 +99,9 @@ def test_get_id(mocker: MockerFixture) -> None:
 def test_unassigned_family_id(mocker: MockerFixture) -> None:
     """Test unassigned_family_id was created and populated."""
     user = add_mock_user(mocker, userAlbert)
-    family_id = user.unassigned_family_id
-    assert family_id
-
-    family = Family.lookup_family(family_id)
+    family = user.unassigned_family_id.fetch()
```
Comment (EpicPi) [inline server/tests/model/test_user.py:102]: can we wrap this call in a method so it doesnt look like you're plopping an id into a family variable?

Comment (EpicPi) [reply server/tests/model/test_user.py:102]: user.get_unassigned_family()
or
a @property decorated
user.unassigned_family

**Review by EpicPi — CHANGES_REQUESTED** [review]
few small things

File: server/tests/model/test_user.py (hunk 3 above)
Comment (SammyBaek) [reply server/tests/model/test_user.py:102]: good idea, I'll refactor

File: server/tests/model/test_user.py [hunk 4]
```diff
@@ -100,12 +100,9 @@ def test_get_id(mocker: MockerFixture) -> None:
...
     """Test unassigned_family_id was created and populated."""
     user = add_mock_user(mocker, userAlbert)
-    family_id = user.unassigned_family_id
-    assert family_id
-
-    family = Family.lookup_family(family_id)
+    family = user.unassigned_family
     assert family
-    assert family.get_id() == str(family_id)
+    assert family.get_id() == str(family.id)
```
Comment (EpicPi) [inline server/tests/model/test_user.py:105]:
```suggestion
```

**Review by EpicPi — APPROVED** [review]
Approved with nit

**Pull request merged**